### PR TITLE
Inline jira:: macro

### PR DESCRIPTION
**Pull request by osfameron**
Instead of having to write something like the following in release
notes:

 * https://issues.couchbase.com/browse/PYCBC-849[PYCBC-849]:

Now we can use a macro:

 * jira::PYCBC-849[]

Full usage details and configuration in the supplied `extensions.adoc`
(not added to nav for now, as it's internal. Is there a better place
to put this?)

NB: the configuration is currently done inside the asciidoc extension,
as these don't accept config.
A better solution might be to convert this to an *Antora* extension
(which does accept config) which simply reads the config, and registers
a properly configured asciidoc extension dynamically?

### CODE REVIEW

**Comment by RayOffiah** [comment]
I like it! Great idea, especially for the Release Notes page.

It feels like it should be an Antora extension though, at least as some sort of wrapper so that the config can go in the playbook. I'm not too familiar with Antora extensions yet, so I'm not sure. I would probably hold off until we get a steer from @mojavelinux. 

But this would be really useful, definitely.

Oh, and is it possible to have something like 

`jira::AV-8828[this Jira]`

**Comment by RayOffiah** [comment]
But this is a really good general purpose extension, so I would stick to making it a general purpose AsciiDoc module with an Antora wrapper (if possible) to get the params from the playbook.

**Comment by osfameron** [comment]
> jira::AV-8828[this Jira]
@RayOffiah yeah, I wondered about that... not sure if it should *replace* the default text (the ticket number) or end up with something like "this Jira (AV-8828)" ?

Antora wrapper sounds good, I have *an idea*, not sure if it'll work, will timebox 30 mins, and then over to Dan for more guidance :)

**Comment by RayOffiah** [comment]
I was thinking that if you had:

`jira::AV-8828[]`

The you'd just end up with the ticket number. But if they put text in the brackets, then you should just put whatever is in the brackets as the link text, even if they don't include the actual ticket number in the text.

Yup, 30mins sounds reasonable :-)

**Comment by osfameron** [comment]
@RayOffiah OK, the `[]` change should be simple, I can do that!

**Comment by RayOffiah** [comment]
LGTM 👍🏾, but probably wait for Dan to take a look.

**Review by RayOffiah — APPROVED** [review]

File: home/modules/contribute/pages/extensions.adoc [hunk 1]
```diff
@@ -0,0 +1,39 @@
+= Site Extensions
+
+Test and document your Asciidoctor.js or Antora extensions here.
+
+== inline JIRA macro
```
Comment (maria-robobug) [inline home/modules/contribute/pages/extensions.adoc:5]:
```suggestion
== Inline JIRA macro
```

File: home/modules/contribute/pages/extensions.adoc [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+[source,asciidoc]
+----
+== Fixed issues:
+
+* jira:JSCBC-1234[]: reversed the polarity of the neutron flow
+* jira:AV-2345[]: frobnicated the gostaks
+* jira:DOC-999[The dreaded DOC-999 ticket...]: implemented jira macro
+----
+
+[,yml]
```
Comment (maria-robobug) [inline home/modules/contribute/pages/extensions.adoc:18]:
```suggestion
[source,yml]
```

**Review by maria-robobug — APPROVED** [review]
👍 LGTM, some minor comments. 
We should add this change in the staging branch too.

File: home/modules/contribute/pages/extensions.adoc (hunk 2 above)
Comment (mojavelinux) [reply home/modules/contribute/pages/extensions.adoc:18]: FYI: The `source` style is now inferred if a language is specified.

Comment (osfameron) [reply home/modules/contribute/pages/extensions.adoc:18]: Good to know - though I'm happy with keeping it explicit personally, let's discuss within the team.

**Pull request merged**